packages/cli/src/databases/mysqldb/migrations/1636626154933-CreateUserManagement.ts

```diff
@@ -0,0 +1,158 @@
+				'PRIMARY KEY (`id`), ' +
+				'UNIQUE KEY `UQ_5b49d0f504f7ef31045a1fb2eb8` (`scope`,`name`) ' +
+			');'
+		);
+
+		await queryRunner.query(
+			'CREATE TABLE `' + tablePrefix + 'user` ( ' +
+				'`id` VARCHAR(100) NOT NULL, ' +
+				'`email` VARCHAR(254) NULL DEFAULT NULL, ' +
+				'`firstName` VARCHAR(32) NULL DEFAULT NULL, ' +
+				'`lastName` VARCHAR(32) NULL DEFAULT NULL, ' +
+				'`password` VARCHAR(200) NULL DEFAULT NULL, ' +
+				'`resetPasswordToken` VARCHAR(200) NULL DEFAULT NULL, ' +
+				'`resetPasswordTokenExpiration` INT NULL DEFAULT NULL, ' +
+				'`personalizationAnswers` VARCHAR(200) NULL DEFAULT NULL, ' +
+				'`createdAt` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP, ' +
+				'`updatedAt` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP, ' +
+				'`globalRoleId` INT NOT NULL, ' +
+				'PRIMARY KEY (`id`), ' +
+				'UNIQUE INDEX `IDX_e12875dfb3b1d92d7d7c5377e2` (`email` ASC) VISIBLE, ' +
+				'INDEX `FK_f0609be844f9200ff4365b1bb3d_idx` (`globalRoleId` ASC) VISIBLE, ' +
+				'CONSTRAINT `FK_f0609be844f9200ff4365b1bb3d` ' +
+					'FOREIGN KEY (`globalRoleId`) ' +
+					'REFERENCES `n8n`.`role` (`id`) ' +
+					'ON DELETE NO ACTION ' +
+					'ON UPDATE NO ACTION);'
+		);
+
+		await queryRunner.query(
+		  'CREATE TABLE `' + tablePrefix + 'shared_workflow` ( ' +
+				'`createdAt` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP, ' +
+				'`updatedAt` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP, ' +
+				'`roleId` INT NOT NULL, ' +
+				'`userId` VARCHAR(100) NOT NULL, ' +
+				'`workflowId` INT NOT NULL, ' +
+				'INDEX `FK_3540da03964527aa24ae014b780_idx` (`roleId` ASC) VISIBLE, ' +
+				'INDEX `FK_82b2fd9ec4e3e24209af8160282_idx` (`userId` ASC) VISIBLE, ' +
+				'INDEX `FK_b83f8d2530884b66a9c848c8b88_idx` (`workflowId` ASC) VISIBLE, ' +
+				'PRIMARY KEY (`userId`, `workflowId`), ' +
+				'CONSTRAINT `FK_3540da03964527aa24ae014b780` ' +
+					'FOREIGN KEY (`roleId`) ' +
+					'REFERENCES `' + tablePrefix + 'role` (`id`) ' +
+					'ON DELETE NO ACTION ' +
+					'ON UPDATE NO ACTION, ' +
+				'CONSTRAINT `FK_82b2fd9ec4e3e24209af8160282` ' +
+					'FOREIGN KEY (`userId`) ' +
+					'REFERENCES `' + tablePrefix + 'user` (`id`) ' +
+					'ON DELETE CASCADE ' +
+					'ON UPDATE NO ACTION, ' +
+				'CONSTRAINT `FK_b83f8d2530884b66a9c848c8b88` ' +
+					'FOREIGN KEY (`workflowId`) ' +
+					'REFERENCES `' + tablePrefix + 'workflow_entity` (`id`) ' +
+					'ON DELETE CASCADE ' +
+					'ON UPDATE NO ACTION);'
+		);
+
+		await queryRunner.query(
+			'CREATE TABLE `' + tablePrefix + 'shared_credentials` ( ' +
+			'`createdAt` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP, ' +
+			'`updatedAt` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP, ' +
+			'`roleId` INT NOT NULL, ' +
+			'`userId` VARCHAR(100) NOT NULL, ' +
+			'`credentialsId` INT NOT NULL, ' +
+			'INDEX `FK_c68e056637562000b68f480815a_idx` (`roleId` ASC) VISIBLE, ' +
+			'INDEX `FK_484f0327e778648dd04f1d70493_idx` (`userId` ASC) VISIBLE, ' +
```

Comment:
Thanks. Added for the sake of completeness; in fact, when you create a foreign key constraint on MySQL you automatically gain an index because it is necessary.